packages/caravan-health/src/privacy.ts

```diff
@@ -0,0 +1,335 @@
+      default:
+        throw new Error("Invalid spend type");
+    }
+  }
+
+  /*
+    Name : Topology Score
+
+    Definition :
+      The score is calculated based on the number of inputs and outputs which 
+      influence the topology type of the transaction.
+
+    Calculation :
+      We have 5 categories of transaction type each with their own impact on privacy score
+      - Perfect Spend (1 input, 1 output)
+      - Simple Spend (1 input, 2 outputs)
+      - UTXO Fragmentation (1 input, more than 2 standard outputs)
+      - Consolidation (more than 1 input, 1 output)
+      - CoinJoin or Mixing (more than 1 input, more than 1 output)
+  */
+  getTopologyScore(transaction: Transaction): number {
+    const numberOfInputs: number = transaction.vin.length;
+    const numberOfOutputs: number = transaction.vout.length;
+
+    const spendType: SpendType = this.determineSpendType(
+      numberOfInputs,
+      numberOfOutputs,
+    );
+    const score: number = this.getSpendTypeScore(
+      spendType,
+      numberOfInputs,
+      numberOfOutputs,
+    );
+
+    if (spendType === SpendType.Consolidation) {
+      return score;
+    }
+    for (let output of transaction.vout) {
+      let address = output.scriptPubkeyAddress;
+      let isResued = this.isReusedAddress(address);
+      if (isResued === true) {
+        return score;
+      }
+    }
+    return score * DENIABILITY_FACTOR;
+  }
+
+  /*
+    Name : Mean Transaction Topology Privacy Score (MTPS)
+
+    Definition :
+      The mean topology is evaluated for entire wallet history based on 
+      the tx toplogy score for each transaction. It signifies how well the 
+      transactions were performed to maintain privacy.
+    
+    Calculation :
+      The mean topology score is calculated by evaluating the topology score for each transaction.
+
+    Expected Range : [0, 0.75]
+    -> Very Poor : [0, 0.15]
+    -> Poor : (0.15, 0.3]
+    -> Moderate : (0.3, 0.45]
+    -> Good : (0.45, 0.6]
+    -> Very Good : (0.6, 0.75)
+  */
+  getMeanTopologyScore(transactions: Transaction[]): number {
+    let privacyScore = 0;
+    for (let tx of transactions) {
+      let topologyScore = this.getTopologyScore(tx);
+      privacyScore += topologyScore;
+    }
+    return privacyScore / transactions.length;
+  }
+
+  /*
+    Name : Address Reuse Factor (ARF)
+
+    Definition :
+      The address reuse factor is evaluates the amount being held by reused addresses with respect 
+      to the total amount. It signifies the privacy health of the wallet based on address reuse.
+
+    Calculation : 
+      The factor is calculated by summing the amount held by reused addresses and dividing it 
+      by the total amount.
+
+    Expected Range : [0,1]
+    -> Very Poor : (0.8, 1]
+    -> Poor : [0.6, 0.8)
+    -> Moderate : [0.4, 0.6)
+    -> Good : [0.2, 0.4)
+    -> Very Good : [0 ,0.2) 
+  */
+  addressReuseFactor(
+    utxos: AddressUtxos,
+    addressUsageMap: AddressUsageMap,
```

Comment:
Wouldn't these also be set on the class. If we imagine these scoring classes as keeping track of the state of a particular wallet instance, then we're just calling different methods to get the scores for that wallet. So you setup the "wallet" with txs and utxos (this is kind of all there is in terms of the statefulness of a wallet), and then would calculate internally the address usage map which would get saved in the class as well. This would mean that you can just call `myPrivacyScorer.addressReuseFactor()`. If we wanted to support more flexibilility you could either make the args optional or support updating the utxos and txs with some sort of class setter, but I don't think that's really necessary just yet.